test/src/unit-bson.cpp

```diff
@@ -48,46 +48,53 @@ TEST_CASE("BSON")
         SECTION("null")
         {
             json j = nullptr;
-            REQUIRE_THROWS_AS(json::to_bson(j), json::type_error);
+            REQUIRE_THROWS_AS(json::to_bson(j), json::type_error&);
+            CHECK_THROWS_WITH(json::to_bson(j), "[json.exception.type_error.317] JSON value of type 0 cannot be serialized to requested format");
```

Comment:
I don't think "type 0" etc. is a helpful description for users of the library. Please use `type_name()` to explicitly name the type (null, boolean, etc.).

Furthermore, "to requested format" should be replaced by "BSON" to be explicit.